doc/blog/201809-UnoUnderTheHoodXamlGeneration.md

```diff
@@ -0,0 +1,84 @@
+            Content =             new global::Windows.UI.Xaml.Controls.StackPanel 
+            { 
+                // Source ..\..\..\..\..\..\UnoExtTestbed.Shared\MainPage.xaml (Line 8:3) 
+                Children =  
+                { 
+                    new global::Windows.UI.Xaml.Controls.TextBlock 
+                    { 
+                        Name = "ClickTextBlock", 
+                        Text = "Button wasn't clicked yet", 
+                        // Source ..\..\..\..\..\..\UnoExtTestbed.Shared\MainPage.xaml (Line 9:4) 
+                    } 
+                    .MainPage_05a03e67d6a6581089c5d12899278d61_XamlApply((MainPage_05a03e67d6a6581089c5d12899278d61XamlApplyExtensions.XamlApplyHandler0)(c0 =>  
+                    { 
+                        nameScope.RegisterName("ClickTextBlock", c0); 
+                        this.ClickTextBlock = c0; 
+                    } 
+                    )) 
+                    , 
+                    new global::Windows.UI.Xaml.Controls.Button 
+                    { 
+                        Content = @"Click me", 
+                        // Source ..\..\..\..\..\..\UnoExtTestbed.Shared\MainPage.xaml (Line 11:4) 
+                    } 
+                    .MainPage_05a03e67d6a6581089c5d12899278d61_XamlApply((MainPage_05a03e67d6a6581089c5d12899278d61XamlApplyExtensions.XamlApplyHandler1)(c1 =>  
+                    { 
+                        c1.Click += Button_Click; 
+                    } 
+                    )) 
+                    , 
+                } 
+            }
+````
+
+What a mouthful! In the normal course of Uno development you don't need to look at code generated from XAML or even know that it's there, though occasionally it's useful to take a peek when debugging.
+
+Recall that the `Button_Click` event is defined in [the code-behind](https://medium.com/@unoplatform/pushing-the-right-buttons-how-uno-implements-views-under-the-hood-a5e93ea86688#the-number-goes-up) for MainPage. As far as the compiler is concerned, both the authored and generated files are just [partial definitions](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/partial-classes-and-methods) of the same class. 
+
+One last fun fact: when building on MacOS using [Visual Studio for Mac](https://visualstudio.microsoft.com/vs/mac/) (which Uno tentatively supports in preview), Uno uses an internal port in place of the System.Xaml namespace (which isn't available on Mac). The same code backs Uno's support for runtime Xaml interpretation via [Windows.UI.Xaml.Markup.XamlReader](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.markup.xamlreader), which is used in the [interactive mode](https://github.com/nventive/Uno.Playground/blob/master/src/Uno.Playground.Shared/Samples/Playground.xaml) of the [Uno Gallery app](https://github.com/nventive/Uno.Playground) and [Uno.Playground website](https://playground.platform.uno/). 
```

Comment:
There are many pieces in parenthesis in this sentence, I think we can avoid some. I would suggest " [...] Uno uses an internal port of the System.Xaml namespace because it isn't available on Mac"